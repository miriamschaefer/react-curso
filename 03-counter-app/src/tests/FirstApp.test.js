import React from 'react';
// import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import FirstApp from '../FirstApp';

describe('Tests in FirstApp component', () => {

    // NOTAS

    // test('Should return greeting message', () => {
    //     const greeting = 'Hi';
    //     // The previous line is used to apply destructuring so we can pass getByText method to our wrapper variable
    //     // const wrapper = render(<FirstApp greeting={greeting}/>);

    //     const { getByText } = render(<FirstApp greeting={ greeting }/>);
    //     expect(getByText(greeting)).toBeInTheDocument();
    // });

    test('should show <FirstApp /> correctly', () => {
        const greeting = 'Hi';
        const wrapper = shallow(<FirstApp greeting={greeting}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should render the subtitle props', () => {
        const greeting = 'Hi';
        const subtitle = ':)';
        const wrapper = shallow(
            <FirstApp
                greeting={greeting}
                subtitle={subtitle}
            />
        );

        const paragraph = wrapper.find('p').text();

        expect(paragraph).toBe(subtitle);
    })
});