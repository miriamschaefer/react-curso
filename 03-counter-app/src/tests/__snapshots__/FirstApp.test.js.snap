// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in FirstApp component should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hi
  </h1>
  <p>
    :)
  </p>
</Fragment>
`;
